Food_Tracker_App/app/src/main/java/com/example/foodtracker/ui/recipes/RecipesMainScreen.java

```diff
@@ -88,6 +95,25 @@ private void addRecipe(Recipe recipe) {
                 adapter.notifyItemInserted(recipeArrayList.indexOf(recipe)));
     }
 
+    public void deleteRecipe(Recipe recipe) {
+        int index = 0;
+        for (index = 0; index < recipeArrayList.size(); index++) {
+            if (Objects.equals(recipeArrayList.get(index).getTitle(), recipe.getTitle())
+                    && Objects.equals(recipeArrayList.get(index).getPrepTime(), recipe.getPrepTime())
+                    && Objects.equals(recipeArrayList.get(index).getServings(), recipe.getServings())
+                    && Objects.equals(recipeArrayList.get(index).getCategory(), recipe.getCategory())
+                    && Objects.equals(recipeArrayList.get(index).getComment(), recipe.getComment())
+                    && Objects.equals(recipeArrayList.get(index).getImage(), recipe.getImage())
```

Comment:
Should override the equals class in `Recipe` the only thing that you should really need to check is that the keys are equal, everything else is determined by the key.

Reply:
I tried overriding the equals method before and it wasn't working, which is why I switched to doing it this way. I tried overriding the equals method again and comparing the keys, but it still doesn't work. I will commit the code anyway so you can see what I did. (I also tried overriding the hashcode method to see if that was causing the issue, but it still didn't work).
This is the error I'm gettting:
java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference at com.example.foodtracker.model.recipe.Recipe.equals(Recipe.java:96)

Reply:
I'll take a quick look!

Reply:
I pushed a fix to this, the problem was that the superclass did not also implement serializable, so I switched this around. So that the superclass implements it and the subclasses implement it by default through `Document`, now we can compare using the document key!